### PR TITLE
Restrict looker-sdk version 24.18.0 and microsoft-kiota-http 1.3.4

### PR DESCRIPTION
**Pull request by gopidesupavan**
The latest looker-sdk version causing import problems in our CI. so restricting 24.18.0 looker-sdk version.

https://github.com/apache/airflow/actions/runs/11295803154/job/31419668010?pr=42950#step:14:1642 

There issue already opened here: https://github.com/looker-open-source/sdk-codegen/issues/1518

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jscheffl — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]

**Comment by gopidesupavan** [comment]
microsoft provider tests are failing due to import errors, its because of the latest microsoft-kiota-http release. and in microsoft provider using msgraph_core (this internally uses kiota-http) causing issues.

**Comment by gopidesupavan** [comment]
Is there any place we can restrict the pip versions which are indirectly installed, in this case kiota-http is using inside msgraph_core, how to restrict pip version?

**Comment by gopidesupavan** [comment]
not sure whether update microsoft-kiota-http pip version works or not , but i am trying out.

**Comment by potiuk** [comment]
> Is there any place we can restrict the pip versions which are indirectly installed, in this case kiota-http is using inside msgraph_core, how to restrict pip version?

It's a bit case-by-case.

Usually it is ok to add such transitive dependency directly to provider as additional dependency - we've done that a number of times in the past - also it is often accompanied by an issue to the upstream project to let them know they have incompatibility and suggest to add similar transitive dependency in their package.  What you've done here is following that and is a valid approach.

However, often it turns out, that such "transitive dependency" limit is already added in a newer version of our direct dependency - but only in one of the latest versions and `uv` or `pip` do not resolve the latest version automatically - in such case we can also bump minimum version of such direct dependency. But this is really something that rquires to take a close look at history of changes in dependencies of such dependencies and understanding them.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks for the fix @gopidesupavan BTW :)

**Comment by gopidesupavan** [comment]
Thanks @potiuk  😄 , updated the issue ids to provider yaml.

**Comment by gopidesupavan** [comment]
mypy-provider check for appflow not satisfying, not sure might be cache issue..

**Comment by gopidesupavan** [comment]
Can this be merged please? One test is failing, it looks like flaky test, will have a look and check-in separate pr

File: providers/src/airflow/providers/amazon/aws/hooks/appflow.py [hunk 1]
```diff
@@ -117,9 +117,9 @@ def update_flow_filter(self, flow_name: str, filter_tasks, set_trigger_ondemand:
...
             flowName=response["flowName"],
-            destinationFlowConfigList=response["destinationFlowConfigList"],
-            sourceFlowConfig=response["sourceFlowConfig"],
-            triggerConfig=response["triggerConfig"],
+            destinationFlowConfigList=response["destinationFlowConfigList"],  # type: ignore[arg-type]
+            sourceFlowConfig=response["sourceFlowConfig"],  # type: ignore[arg-type]
+            triggerConfig=response["triggerConfig"],  # type: ignore[arg-type]
             description=response.get("description", "Flow description."),
-            tasks=tasks,
+            tasks=tasks,  # type: ignore[arg-type]
```
Comment (eladkal) [inline providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: How is this change related to the PR?

Comment (gopidesupavan) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: ah, actually there was mypy checks failed in this pr and flagged above things, so i have updated them part of this. https://github.com/apache/airflow/actions/runs/11300287987/job/31433030516.

May be i should have done that in different pr, sorry for the confusion..

Comment (eladkal) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: i am not sure if we should just add ignore or fix the actual issue cause it.
We need to check why this was cause failures

Comment (eladkal) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: cc @ferruzzi @vincbeck

Comment (potiuk) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: Yeah. Likely fix would be better :)

Comment (gopidesupavan) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: Agree, looking into the latest release of mypyboto3 appflow has updates related to this. Will update the changes :)

Comment (eladkal) [reply providers/src/airflow/providers/amazon/aws/hooks/appflow.py:124]: https://github.com/apache/airflow/pull/43436
